### PR TITLE
[FLINK-10174][table] fix getbytes in hex and toBase64

### PR DESCRIPTION
**Pull request by xueyumusic**
## What is the purpose of the change
Hex and toBase64 builtin method use str.getBytes() with no Charset. It maybe depend on local execution environment for special Unicode and maybe result in errors when test Hex for special Unicode

## Brief change log

using str.getBytes(Charsets.UTF_8) instead of str.getBytes()

## Verifying this change
add tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala [hunk 1]
```diff
@@ -211,7 +212,7 @@ object ScalarFunctions {
   /**
     * Returns the base64-encoded result of the input string.
     */
-  def toBase64(base: String): String = Base64.encodeBase64String(base.getBytes())
+  def toBase64(base: String): String = Base64.encodeBase64String(base.getBytes(Charsets.UTF_8))
```
Comment (twalthr) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala]: Can you also add the charset to `fromBase64`?

File: flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala [hunk 2]
```diff
@@ -563,6 +570,13 @@ class ScalarFunctionsTest extends ScalarTypesTestBase {
       "f33.fromBase64()",
       "FROM_BASE64(f33)",
       "null")
+
+    testAllApis(
+      "5L2g5aW9".fromBase64(),
```
Comment (twalthr) [inline flink-libraries/flink-table/src/test/scala/org/apache/flink/table/expressions/ScalarFunctionsTest.scala]: Actually, this looks correct for me. It's weird that MySQL returns `e4bda0e5a5bd` for this.

File: flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala [hunk 3]
```diff
@@ -221,7 +222,7 @@ object ScalarFunctions {
   /**
     * Returns the hex string of a string argument.
     */
-  def hex(x: String): String = Hex.encodeHexString(x.getBytes).toUpperCase()
+  def hex(x: String): String = Hex.encodeHexString(x.getBytes(Charsets.UTF_8)).toUpperCase()
```
Comment (twalthr) [inline flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala]: @xueyumusic please enable more IDE checkstyles. `Charsets` is deprecated. I will fix this while merging.

Comment (xueyumusic) [reply flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/functions/ScalarFunctions.scala]: Thanks for the suggestion! @twalthr . It looks this place https://github.com/apache/flink/blob/master/flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/calls/HashCalcCallGen.scala#L71
also used Charsets.UTF_8, maybe could update at the same time.

**Pull request closed without merging**